examples/find_all_posts_mentioning_github.py

```diff
@@ -1,25 +1,35 @@
-import twingly_search
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
 import datetime
 
+import twingly_search
+
+
 class SearchPostStream:
     def __init__(self, keyword, language=''):
         self.client = twingly_search.Client(user_agent="MyCompany/1.0")
-        self.query = self.client.query()
-        self.query.pattern = "sort-order:asc sort:published %s" % keyword
-        self.query.start_time = datetime.datetime.utcnow() - datetime.timedelta(hours=48)
-        self.query.language = language
+        self.q = self.client.query()
+        self.q.search_query = "%s sort-order:asc sort:published " % keyword
+        self.q.language = language
+        self.q.start_date = datetime.datetime.utcnow() - datetime.timedelta(weeks=10)
+        self.keyword = keyword
 
     def each(self):
         while True:
-            result = self.query.execute()
+            query_string = self.q.build_query_string()
+            result = self.client.execute_query(query_string)
```

Comment:
Yes, we have started to realised that as well :) (it's the same problem in a couple of other clients).

What if we do something along the lines of https://github.com/twingly/twingly-search-api-python/pull/32/files/1e3a4f6b18b4b043e227c6c4020d827720f1104f#r117175215 (let the client call `q.build_query_string()` if the argument to `client.execute_query(q)` is of type `Query` that is). That way the user doesn't have to call `query.build_query_string()` themselves.